lib/processors/themeBuilderWorker.js

```diff
@@ -91,3 +72,145 @@ if (!workerpool.isMainThread) {
 		execThemeBuild
 	});
 }
+
+
+export class FsMainThreadInterface {
+	#comPorts = new Set();
+	#fsInterfaceReader = null;
+	#cache = Object.create(null);
+
+	constructor(fsInterfaceReader) {
+		if (!fsInterfaceReader) {
+			throw new Error("fsReader is mandatory argument");
+		}
+
+		this.#fsInterfaceReader = fsInterfaceReader;
+	}
+
+	/**
+	 * Adds MessagePort and starts listening for requests on it.
+	 *
+	 * @param {MessagePort} comPort port1 from a {code}MessageChannel{/code}
+	 */
+	startCommunication(comPort) {
+		if (!comPort) {
+			throw new Error("Communication channel is mandatory argument");
+		}
+
+		this.#comPorts.add(comPort);
+		comPort.on("message", (e) => this.#onMessage(e, comPort));
+		comPort.on("close", () => comPort.close());
+	}
+
+	/**
+	 * Ends MessagePort communication.
+	 *
+	 * @param {MessagePort} comPort port1 to remove from handling.
+	 */
+	endCommunication(comPort) {
+		comPort.close();
+		this.#comPorts.delete(comPort);
+	}
+
+	/**
+	 * Destroys the FsMainThreadInterface
+	 */
+	cleanup() {
+		this.#comPorts.forEach((comPort) => comPort.close());
+		this.#cache = null;
+		this.#fsInterfaceReader = null;
+	}
+
+	#onMessage(e, comPort) {
+		switch (e.what) {
+		case "readFile":
+			this.#doRequest(comPort, {what: "readFile", fsPath: e.fsPath, options: e.options});
+			break;
+		case "stat":
+			this.#doRequest(comPort, {what: "stat", fsPath: e.fsPath});
+			break;
+		case "readdir":
+			this.#doRequest(comPort, {what: "readdir", fsPath: e.fsPath});
+			break;
+		case "mkdir":
+			this.#doRequest(comPort, {what: "mkdir", fsPath: e.fsPath});
+			break;
+		}
+	}
+
+	async #doRequest(comPort, {what, fsPath, options}) {
+		const cacheKey = [fsPath, what].join("-");
+		if (!this.#cache[cacheKey]) {
+			this.#cache[cacheKey] = new Promise((res) => {
+				if (what === "readFile") {
+					this.#fsInterfaceReader.readFile(fsPath, options, (error, result) => res({error, result}));
+				} else {
+					this.#fsInterfaceReader[what](fsPath, (error, result) => res({error, result}));
+				}
+			});
+		}
+
+		const fromCache = await this.#cache[cacheKey];
+		comPort.postMessage({what, fsPath, ...fromCache});
+	}
+}
+
+export class FsWorkerThreadInterface {
+	#comPort = null;
+	#callbacks = [];
+	#cache = Object.create(null);
+
+	constructor(comPort) {
+		if (!comPort) {
+			throw new Error("Communication port is mandatory argument");
+		}
+
+		this.#comPort = comPort;
+		this.#comPort.on("message", this.#onMessage.bind(this));
+		this.#comPort.on("close", this.#onClose.bind(this));
+	}
+
+	#onMessage(e) {
+		const cbObject = this.#callbacks.find((cb) => cb.what === e.what && cb.fsPath === e.fsPath);
```

Comment:
At first reading I was not able to guess the meaning of `what` in here. Further down in the code it becomes clear, that it is the fs function name to be invoked.

Reply:
I've renamed it to "action". Maybe, makes more sense